pyflagser/flagser.py

```diff
@@ -1,40 +1,150 @@
+           documentation_flagser.pdf>`_.
+
+    """
+    # Handle default parameters
+    if max_dimension == np.inf:
+        _max_dimension = -1
+    else:
+        _max_dimension = max_dimension
+
+    if approximation is None:
+        _approximation = -1
+    else:
+        _approximation = approximation
+
+    # All edge filtrations are equivalent in the static case
+    _filtration = 'max'
+
+    # Extract vertices and edges
+    vertices, edges = _extract_unweighted_graph(adjacency_matrix)
+
+    # Call flagser binding
+    homology = compute_homology(vertices, edges, min_dimension, _max_dimension,
+                                directed, coeff, _approximation, _filtration)
+
+    # Creating dictionary of return values
+    out = dict()
+    out['cell_count'] = homology[0].get_cell_count()
+    out['betti'] = homology[0].get_betti_numbers()
+    out['euler'] = homology[0].get_euler_characteristic()
+    return out
+
+
+def flagser_weighted(adjacency_matrix, max_edge_length=None, min_dimension=0,
+                     max_dimension=np.inf, directed=True, filtration="max",
+                     coeff=2, approximation=None):
     """Compute persistent homology of a directed/undirected
     weighted/unweighted flag complexes.
 
-    Important: the input graphs cannot contain self-loops, i.e. edges
-    that start and end in the same vertex, therefore diagonal elements
-    of the flag matrix store vertex weights.
+    From an `adjacency_matrix` and a `filtration` construct a filtered
+    flag complex as a sequence of its cells associated to their filtration
+    values and compute its persistent homology.
 
     Parameters
     ----------
-    flag_matrix : ndarray or scipy.sparse matrix, required
+    adjacency_matrix : 2d ndarray or scipy.sparse matrix, required
         Matrix representation of a directed/undirected weighted/unweighted
         graph. Diagonal elements are vertex weights. The way zero values are
         handled depends on the format of the matrix. If the matrix is a dense
-        ``np.ndarray``, all zeros are explicitly accounted for and denote
-        zero-weight edges, i.e., edges appearing at filtration value zero.
-        If the matrix is a sparse ``scipy.sparse`` matrix, zeros on the
-        diagonal and off-diagonal zeros assigned directly are treated
-        explicitly. Off-diagonal zeros that have not been assigned directly are
-        treated implicitly, i.e., correspond to an absent edge.
+        ``np.ndarray``, zero values denote zero-weighted edges. If the matrix
+        is a sparse ``scipy.sparse`` matrix, explicitely stored off-diagonal
+        zeros  and all diagonal zeros denote zero-weighted edges. Off-diagonal
+        values that have not been explicitely stored are treated by
+        ``scipy.sparse`` as zeros but will be understood as infinitely-valued
+        edges, i.e., edges absent from the filtration.
+
+    max_edge_length : int or float or ``None``, optional, default: ``None``
```

Comment:
Not sure I'd introduce the word "length" here. Unlike Vietoris-Rips, here we are less tied to the metric picture where edge weights are distances/lengths, and I think our terminology should be more general accordingly. How about `max_edge_weight` instead?

Reply:
Makes sense,thank you!